packages/cli/src/push/abstract.push.ts

```diff
@@ -1,17 +1,29 @@
-import { jsonStringify } from 'n8n-workflow';
+import { EventEmitter } from 'events';
+import { assert, jsonStringify } from 'n8n-workflow';
 import type { IPushDataType } from '@/Interfaces';
 import { Logger } from '@/Logger';
+import type { User } from '@/databases/entities/User';
 
-export abstract class AbstractPush<T> {
+/**
+ * Abstract class for two-way push communication.
+ * Keeps track of user sessions and enables sending messages.
+ *
+ * @emits message when a message is received from a client
+ */
+export abstract class AbstractPush<T> extends EventEmitter {
```

Comment:
This class abstracts away the details of how we push from the backend to the frontend. One of the backends here is SSE, which can not support bi-directional communication.

Perhaps we should either
* rename this class to something else that indicates that this isn't just for "push" now, and add a warning when SSE backend is in use, or
* leave the existing push code as it is, and setup a second  mechanism for bidirectional communication that explicitly uses websockets.

Reply:
I was thinking whether to create a separate mechanism for two-way communication, but since this feature is not critical it's okay if it doesn't work when n8n is configured to use SSE. So I'd opt for the first option. I was thinking of renaming it to `AbstractTwoWayPush` to indicate that it works in pushing both directions. WDYT?

The second option would require more work. Also not sure how confusing it is if you can configure the server to use SSE but then we use websockets nonetheless? Why don't we then just use websocket for everything?

Reply:
We only recently added websockets, and made it the default only in v1. SSE is deprecated, but still around because when users complain about issues with websockets, we ask to to switch back to SSE while we figure out the issue. We still have users who have issues with websockets behind certain reverse-proxies.

Reply:
@netroy ready for re-review now